adapters/Chartboost/ChartboostAdapter/GADMAdapterChartboost.m

```diff
@@ -159,15 +188,61 @@ - (void)didFailToLoadInterstitial:(CBLocation)location withError:(CBLoadError)er
 }
 
 - (void)didDismissInterstitial:(CBLocation)location {
-  id<GADMAdNetworkConnector> strongConnector = _interstitialConnector;
+  id<GADMAdNetworkConnector> strongConnector = _connector;
   [strongConnector adapterWillDismissInterstitial:self];
   [strongConnector adapterDidDismissInterstitial:self];
 }
 
 - (void)didClickInterstitial:(CBLocation)location {
-  id<GADMAdNetworkConnector> strongConnector = _interstitialConnector;
+  id<GADMAdNetworkConnector> strongConnector = _connector;
   [strongConnector adapterDidGetAdClick:self];
   [strongConnector adapterWillLeaveApplication:self];
 }
 
+#pragma mark - Chartboost Banner Delegate Methods
+
+- (void)didCacheAd:(CHBCacheEvent *)event error:(nullable CHBCacheError *)error
+{
+    id<GADMAdNetworkConnector> strongConnector = _connector;
+    if (error) {
+        [strongConnector adapter:self didFailAd:NSErrorWithCHBCacheError(error)];
+    } else {
+        [strongConnector adapter:self didReceiveAdView:(UIView *)event.ad];
+    }
+    // We keep a strong reference to the banner only until it is loaded, since at that point the view is sent to GMA and it is its responsibility to retain it.
+    _loadingBanner = nil;
```

Comment:
Does nilling here also removes the banner delegate? Do we receive ad events even after nilling out? Because in line 134 we added this class as a delegate.

Reply:
It doesn't remove the banner delegate because at this point some other object must have kept a strong reference to the banner. The superview it has been place inside at least.
We receive ad events after nilling it out. The banner delegate is the adapter itself, which should be kept alive, and the banner itself isn't released.
The point in doing this is to keep a reference to the banner only as long as it is necessary to prevent memory leaks.